src/browser/fake_network.js

```diff
@@ -43,6 +58,220 @@ function iptolong(parts) {
     return parts[0] << 24 | parts[1] << 16 | parts[2] << 8 | parts[3];
 }
 
+class Uint8Stream
```

Comment:
I'd say (Growable)Ringbuffer would be a better name.

Reply:
Done, and GrowableRingbuffer it is (this was the initial name of this class).